index.d.ts

```diff
@@ -0,0 +1,108 @@
+declare module 'dicom-parser' {
+ 
+  export type ByteArray = Uint8Array | Buffer;
+
+  export interface Element {
+    tag: string;
+    vr?: string;
+    length: number;
+    dataOffset: number;
+    items?: Element[];
+    dataSet?: DataSet;
+    parser?: ByteArrayParser;
+  }
+
+  export interface DataSet {
+    byteArray: ByteArray;
+    byteArrayParser : ByteArrayParser;
+    elements: {
+      [key: string]: Element;
+    };
+    warnings: string[];
+
+    /**
+     * Finds the element for tag and returns an unsigned int 16 if it exists and has data. Use this function for VR type US.
+     */
+    uint16: (tag: string, index: number) => number;
+
+    /**
+     * Finds the element for tag and returns an signed int 16 if it exists and has data. Use this function for VR type SS.
+     */
+    int16: (tag: string, index: number) => number;
+
+    /**
+     * Finds the element for tag and returns an unsigned int 32 if it exists and has data. Use this function for VR type UL.
+     */
+    uint32: (tag: string, index: number) => number;
+
+    /**
+     * Finds the element for tag and returns an signed int 32 if it exists and has data. Use this function for VR type SL.
+     */
+    int32: (tag: string, index: number) => number;
+
+    /**
+     * Finds the element for tag and returns a 32 bit floating point number if it exists and has data. Use this function for VR type FL.
+     */
+    float: (tag: string, index: number) => number;
+
+    /**
+     * Finds the element for tag and returns a 64 bit floating point number if it exists and has data. Use this function for VR type FD.
+     */
+    double: (tag: string, index: number) => number;
+
+    /**
+     * Returns the Value Multiplicity of an elements - the number of values in a multi-valued element.
+     */
+    numStringValues: (tag: string) => number;
+
+    /**
+     * Finds the element for tag and returns a string if it exists and has data. Use this function for VR types AE, CS, SH, and LO.
+     */
+    string: (tag: string, index: number) => string;
```

Comment:
Sorry guys, can you explain to me why the index parameter is required? Works pretty well without it.

Reply:
Good catch @pavelkeyzik, index should be optional.

I did not make a default parameter in the interface since the implementation doesn't use a default parameter, but rather assigns it to `0` in the function's body.